packages/build-tools/src/context.ts

```diff
@@ -104,10 +106,23 @@ export class BuildContext<TJob extends Job> {
     this.reportError = options.reportError;
     this.metadata = options.metadata;
     this.skipNativeBuild = options.skipNativeBuild;
+    const environmentSecrets: Record<string, string> = {};
+    if (Array.isArray(job?.secrets?.environmentSecrets)) {
+      for (const { name, type, value } of job.secrets.environmentSecrets) {
+        if (type === EnvironmentSecretType.STRING) {
+          environmentSecrets[name] = value;
+        } else {
+          environmentSecrets[name] = createTemporaryEnvironmentSecretFile(
```

Comment:
I'm not sure if this is a good place to create those files, creating context shouldn't have side effects plus doing it here is forcing you to use sync methods